src/test/java/seedu/guest/model/guest/DateRangeTest.java

```diff
@@ -0,0 +1,43 @@
+package seedu.guest.model.guest;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.guest.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class DateRangeTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new DateRange(null));
+    }
+
+    @Test
+    public void constructor_invalidDateRange_throwsIllegalArgumentException() {
+        String invalidDateRange = "";
+        assertThrows(IllegalArgumentException.class, () -> new DateRange(invalidDateRange));
+    }
+
+    @Test
+    public void isValidDateRange() {
+        // null date range
+        assertThrows(NullPointerException.class, () -> DateRange.isValidDateRange(null));
+
+        // invalid date range
+        assertFalse(DateRange.isValidDateRange("")); // empty string
```

Comment:
Should there also be a test case where the date and month values are swapped? Eg 11/29/22

Reply:
Yes, added and clarified comments